lib/init.c

```diff
@@ -11,24 +11,30 @@ int metal_init(const struct metal_init_params *params)
 {
 	int error = 0;
 
-	memset(&_metal, 0, sizeof(_metal));
+	if (_metal.common.ref_count++ == 0) {
+		memset(&_metal.common + 1, 0,
```

Comment:
`&_metal.common + 1 i`s probably not good as` ref_count `is an integer.
More than that I wonder if creating a static variable for refcount would not be more better than having this address shift

Reply:
but is it good to use global reference count for the local variables at least that it isn't a global variable in libmetal level?

Reply:
Ok, I ajdust the change, please take a look.